OpenNefia.Content/CharaMake/CharaMakeFeatWindowLayer.cs

```diff
@@ -90,6 +93,28 @@ public override void Update(float dt)
             base.Update(dt);
             FeatWindow.Update(dt);
         }
+
+        public override void ApplyStep(EntityUid entity)
+        {
+            base.ApplyStep(entity);
+            if (!Data.TryGetValue<Dictionary<PrototypeId<FeatPrototype>, int>>(ResultName, out var feats))
+            {
+                Logger.WarningS("charamake", "No attributes in CharaMakeData");
+                return;
+            }
+
+            if (!_entityManager.TryGetComponent<FeatsComponent>(entity, out var featsComponent))
+            {
+                Logger.WarningS("charamake", "No FeatsComponent present on entity");
+                return;
+            }
+
+            foreach (var feat in feats)
+            {
+                featsComponent.Feats.TryGetValue(feat.Key, out var currentLevel);
```

Comment:
Check the return value if you use `TryGetValue`:

```csharp
if (featsComponent.Feats.TryGetValue(feat.Key, out var currentLevel)) 
{
   // ...
}

Reply:
Usually I'd agree, but we need to call the function no matter what, and if it's false currentLevel is 0, which is what we want in that case anyway.

Reply:
Maybe at least calling `DebugTools.Assert(featsComponent.Feats.TryGetValue()))` at least to make that assumption clear.

Reply:
changed in 6282bad